modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/touchController.js

```diff
@@ -23,11 +23,36 @@
 angular.module('adminNg.controllers')
 .controller('TouchCtrl', ['$scope', '$document',
   function ($scope, $document) {
-
     $scope.openPopup = function(id) {
       var popup = angular.element(document.getElementById(id));
       popup.toggle('is-hidden');
     };
-
+    $scope.clearPopup = function() {
+      var links = document.getElementsByClassName('js-popover');
+      for (var index = 0; index < links.length; index++) {
+        links[index].style.display = 'none';
+      }
+    };
+    window.onclick = function(ev) {
+      if (ev.target.nodeName != 'SPAN') {
+        var links = document.getElementsByClassName('js-popover');
+        for (var index = 0; index < links.length; index++) {
+          links[index].style.display = 'none';
+        }
+      }
+    };
   }
-]);
+])
+.directive('escKey', function() {
```

Comment:
I feel like the directive in this PR and #1073 should be moved into its own file, and then reused, rather than copy and pasted.

Reply:
@gregorydlogan I try to do a quick fix now